code/test/app.js

```diff
@@ -81,54 +70,393 @@ const executeAppTransaction = async ({
+      }
+    )
+    let bobCurrentState = await TestUtil.getState(bob.wallet)
+    // Validate succesful event emitted
+    TestUtil.validateEvent({ tx, expectedEvent: 'ExternalCallCompleted' })
+
+    // check alice and bobs balance
+    await TestUtil.validateTokenBalances({
+      receivers: [alice.wallet.address, bob.wallet.address],
+      tokenTypes: [ONEConstants.TokenType.ERC20, ONEConstants.TokenType.ERC20],
+      tokenContracts: [testerc20, testerc20],
+      tokenAmounts: [[900], [100]]
+    })
+
+    // check Alice's current state
+    state = await TestUtil.validateOpsStateMutation({ wallet: alice.wallet, state })
+    // Alice's spending state lastSpendingInterval has been updated
+    const newSpendingState = await TestUtil.getSpendingStateParsed(alice.wallet)
+    const expectedSpendingState = state.spendingState
+    expectedSpendingState.lastSpendingInterval = newSpendingState.lastSpendingInterval
+    state.spendingState = await TestUtil.validateSpendingStateMutation({ expectedSpendingState, wallet: alice.wallet })
+    await TestUtil.assertStateEqual(state, currentState)
+    // Check Bob nothing has changed except for balances above
+    await TestUtil.assertStateEqual(bobState, bobCurrentState)
   })
 
   // ====== BATCH ======
   // Test batching transactions
-  // Expected result a batch of transactions will be processed
-  it('UP-BASIC-22 BATCH: must be able to process a batch of transactions', async () => {
-    assert.strictEqual(0, 1, 'Under Development')
+  // Expected result will track an ERC20 and then transfer 100 ERC20 to Bob
+  // Following is the structure which we use for ONEUtil.abi.encodeParameters
+  //   struct OperationParams {
+  //     uint8 Enums.OperationType operationType;
+  //     uint8 Enums.TokenType tokenType;
+  //     address contractAddress;
+  //     uint256 tokenId;
+  //     address payable dest;
+  //     uint256 amount;
+  //     bytes data;
+  // }
+  it('AP-BASIC-22 BATCH: must be able to process a batch of transactions', async () => {
+    // Begin Tests
+    let testTime = Date.now()
+
+    // Note here we populate each operation as an object then map it to an array
+    // It was felt this was clearer and allows us to leverage default values however it could be done concisely by providing the values in an array as follows
+    // let operationParams = [ONEConstants.OperationType.TRACK, ONEConstants.TokenType.ERC20, testerc20.address, 0, alice.wallet.address, 1 , new Uint8Array()]
+    let operationParamsArray = []
+    // Alice Tracks the ERC20 token
+    let operationParamsObject = {
+      ...NullOperationParams, // Default all fields to Null values than override
+      operationType: ONEConstants.OperationType.TRACK,
+      tokenType: ONEConstants.TokenType.ERC20,
+      contractAddress: testerc20.address,
+      dest: alice.wallet.address,
+      amount: 1
+    }
+    let operationParams = Object.keys(operationParamsObject).map((key) => operationParamsObject[key])
```

Comment:
Object.values

Reply:
This is currently working not sure that you mean here as to what needs to be changed.
I tried
`    let operationParams = Object.values(operationParamsObject).map((value) => operationParamsObject[value])
`
but this gave
```
  1) Contract: ONEWallet
       AP-BASIC-22 BATCH: must be able to process a batch of transactions:
     TypeError: Cannot read property 'length' of undefined
      at ABICoder.formatParam (node_modules/web3-eth-abi/lib/index.js:223:30)
      at /Users/john/one-wallet/one-wallet/code/node_modules/web3-eth-abi/lib/index.js:112:41
      at Array.forEach (<anonymous>)
      at modifyParams (node_modules/web3-eth-abi/lib/index.js:107:30)
      at /Users/john/one-wallet/one-wallet/code/node_modules/web3-eth-abi/lib/index.js:105:43
      at Array.map (<anonymous>)
      at modifyParams (node_modules/web3-eth-abi/lib/index.js:105:34)
      at /Users/john/one-wallet/one-wallet/code/node_modules/web3-eth-abi/lib/index.js:116:13
      at Array.map (<anonymous>)
      at ABICoder.encodeParameters (node_modules/web3-eth-abi/lib/index.js:93:21)
      at Context.<anonymous> (test/app.js:331:33)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      ```